cli/health.php

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env php
+<?php
+declare(strict_types=1);
+
+if (php_sapi_name() !== 'cli') {
+	echo 'Error: This script may only be invoked from command line!', PHP_EOL;
+	die(2);
+}
+
+$options = getopt('', ['url::', 'connect_timeout::', 'timeout::']);
+$address = is_string($options['url'] ?? null) ? $options['url'] : 'http://localhost/i/';
+$ch = curl_init($address);
+if ($ch === false) {
+	fwrite(STDERR, 'Error: Failed to initialize cURL!' . PHP_EOL);
+	die(3);
+}
+curl_setopt_array($ch, [
+	CURLOPT_CONNECTTIMEOUT => is_numeric($options['connect_timeout'] ?? null) ? (int)$options['connect_timeout'] : 3,
+	CURLOPT_TIMEOUT => is_numeric($options['timeout'] ?? null) ? (int)$options['timeout'] : 5,
+	CURLOPT_ENCODING => '',	//Enable all encodings
+	CURLOPT_HTTPHEADER => [
+		'Connection: close',
+	],
+	CURLOPT_RETURNTRANSFER => true,
+]);
+$content = curl_exec($ch);
+$httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+
+if ($httpCode !== 200 || !is_string($content) || !str_contains($content, 'jsonVars')) {
```

Comment:
Maybe it's obvious to PHP experts, but for me a comment would be helpful on why is it expecting the magic string "jsonVars"?

Reply:
`jsonVars` is just something relatively unique, which is found in FreshRSS HTML responses:

https://github.com/FreshRSS/FreshRSS/blob/e7c9c3d11e9d5291acd31cd667504fba4e80a0c8/app/layout/layout.phtml#L34

This allows reporting an error if another kind of content is served at that address

Reply:
A status code is sent with the headers. If an error should occur along the way after the headers in rare cases you might receive a 200 with what's still a practical failure, in any language. This is fairly decent evidence that the first few hundred bytes should actually be correct in a way that something generic like `</html>` wouldn't be, though @Alkarex it might potentially be worth adding something like that as an additional safety check at the end?

```suggestion
if ($httpCode !== 200 || !is_string($content) || !str_contains($content, 'jsonVars') || !str_contains($content, '</html>')) {
```

I'm not sure if a comment can add much though. It's just a string of an above average uniqueness level that should be present but that's what the code itself already says. I suppose you could add that it's evidence a couple of templating functions already succeeded or something.

Reply:
Ok, thanks. I think I'd add that info as a code comment.